docs/contributing.md

```diff
@@ -0,0 +1,80 @@
+- [Minikube](https://minikube.sigs.k8s.io/docs/)
+- [K3S](https://k3s.io/)
+- [KinD (Kubernetes in Docker)](https://kind.sigs.k8s.io/)
+
+### Keda
+
+Follow the [install instructions](./install.md) to check out how to install and get this add-on up and running.
+
+## Build scripts
+
+This project uses [Mage](https://magefile.org) as opposed to Make because it's way faster to build and push images, as
+well as to run tests and other common tasks.
+
+It's not required to have Mage installed as a dependency because all the files have been prebuilt as binaries
+called [build](../build) located both on the root directory and the [operator](../operator) directory. You can use the
+original Mage functionality if you have the Mage tool installed in your shell.
+
+> **Note:** The build binary located in the root directory is related to the whole project, so it gives you the ability to control the build and install process of all the modules in this project. On the other hand, the build binary located in the [operator](../operator/build) directory, is **just related to the operator module**.
+
+The usage is as follows:
+
+- Type `./build` on the binary directory to print a list of all available commands
+- Type `./build -h <command>` to check the help for that specific command
+- `./build -h` shows the general help
+
+Most of the commands are simple, and we have a few commands that chain other commands together, for reference on chains,
+check the [Magefile](../magefile.go) source code. Below is a list of the most common build commands
+
+> All commands are case insensitive, so `buildAll` and `buildall` are the same.
+
+In the root directory:
+
+- `./build buildAll`: Builds all the binaries for local testing.
+- `./build deleteOperator [namespace]`: Deletes the installed add-on in the given `namespace` for the active K8S
+  cluster.
+- `./build dockerBuildAll <repository>`: Builds all the images for the `interceptor`, `scaler`, and `operator` modules
+  for the specified `repository`.
+    - You can also build specific images by using `./build dockerBuild <repository> <module>`, where module is one
+      of `interceptor`, `scaler`, or `operator`.
+- `./build dockerPushAll <repository>`: Pushes all the built images for a given repository.
+    - You can push the images using `./build dockerPush <repository> <module>` like the `dockerBuild` command.
+- `./build installKeda [namespace]`: will install KEDA on the given namespace.
+- `./build upgradeOperator [namespace] <image>`: Will install the add-on in the given `namespace` if not installed, or
+  update it using the provided `image`.
+
+> The default values for the `namespace` if not provided (when passed as `""`, like `./build upgradeOperator "" image`) is `kedahttp`
+
+In the operator directory:
+
+- `./operator/build Manifests`: Builds all the manifest files for Kubernetes, it's important to build after every change
+  to a Kustomize annotation.
+- `./operator/build All`: Generates the operator.
```

Comment:
could you put these commands in the top-level `build` command, so we have one place to do everything?

Reply:
right away

Reply:
Done